test_conformance/spirv_new/test_op_spec_constant.cpp

```diff
@@ -0,0 +1,182 @@
+    SPIRV_CHECK_ERROR(err, "Failed to create spec_buffer");
+
+    err = clSetKernelArg(kernel, 0, sizeof(clMemWrapper), &numbers_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument inputs_buffer");
+    err = clSetKernelArg(kernel, 1, sizeof(clMemWrapper), &spec_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument spec_constant_value");
+
+    size_t global = num;
+    err = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, &global, NULL, 0, NULL,
+                                 NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to enqueue kernel");
+    clFinish(queue);
+
+    err = clSetProgramSpecializationConstant(prog, 101, sizeof(T),
+                                             &spec_constant_value);
+    SPIRV_CHECK_ERROR(err, "Failed to run clSetProgramSpecializationConstant");
+
+    std::vector<T> device_results(num, 0);
+    err = clEnqueueReadBuffer(queue, numbers_buffer, CL_TRUE, 0, bytes,
+                              device_results.data(), 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to copy from numbers_buffer");
+
+    for (int i = 0; i < num; i++)
+    {
+        if (device_results[i] == references[i])
+        {
+            log_error("Values match but should not at location %d expected %d "
+                      "obtained %d\n",
+                      i, references[i], device_results[i]);
+            return TEST_FAIL;
+        }
+    }
+
+    err = clBuildProgram(prog, 1, &deviceID, NULL, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to build program");
+
+    kernel = clCreateKernel(prog, "spec_const_kernel", &err);
+    SPIRV_CHECK_ERROR(err, "Failed to create kernel");
+
+    err = clSetKernelArg(kernel, 0, sizeof(clMemWrapper), &numbers_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument inputs_buffer");
+    err = clSetKernelArg(kernel, 1, sizeof(clMemWrapper), &spec_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument spec_constant_value");
+
+    err = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, &global, NULL, 0, NULL,
+                                 NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to enqueue cl kernel");
+    clFinish(queue);
+
+    std::fill(device_results.begin(), device_results.end(), 0);
+    err = clEnqueueReadBuffer(queue, numbers_buffer, CL_TRUE, 0, bytes,
+                              device_results.data(), 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to copy from numbers_buffer");
+
+    for (int i = 0; i < num; i++)
+    {
+        if (device_results[i] != references[i])
+        {
+            log_error(
+                "Values do not match in index %d expected %d obtained %d\n", i,
+                references[i], device_results[i]);
+            return TEST_FAIL;
+        }
+    }
+    return TEST_PASS;
+}
+
+
+#define TEST_SPEC_CONSTANT(NAME, type, value, spec_constant_value)             \
+    TEST_SPIRV_FUNC_VERSION(op_spec_constant_##NAME##_simple, Version(2, 1))   \
```

Comment:
Why pass the version since you're always passing the default value? Why was the change to the test registration code needed?

Reply:
This is my typo. This should be Version(2,2). In comparison to other test cases which requires Version(2,1). This is why I need modify registration code (this solves your comment https://github.com/KhronosGroup/OpenCL-CTS/pull/719#discussion_r402922370)